packages/cli-plugin-metro/src/tools/loadMetroConfig.ts

```diff
@@ -89,17 +97,28 @@ export default async function loadMetroConfig(
   // @ts-ignore resolveConfig return value is mistyped
   logger.debug(`Reading Metro config from ${projectConfig.filepath}`);
 
-  try {
-    require.resolve('@react-native/metro-config', {
-      paths: [ctx.root],
-    });
-  } catch (e) {
+  if (
+    !/['"']@react-native\/metro-config['"']/.test(
+      fs
+        // @ts-ignore resolveConfig return value is mistyped
+        .readFileSync(projectConfig.filepath, 'utf8'),
+    )
+  ) {
     logger.warn(
-      "From React Native 0.72, your 'metro.config.js' file should " +
-        "extend '@react-native/metro-config', however it's not present in your " +
-        "project's devDependencies. Please install '@react-native/metro-config'.",
+      'From React Native 0.72, your metro.config.js file should extend' +
+        "'@react-native/metro-config'. Please see the React Native 0.72 " +
+        'changelog, or copy the template at:\n' +
+        'https://github.com/facebook/react-native/blob/main/packages/react-native/template/metro.config.js',
+    );
+    logger.warn('Falling back to internal defaults.');
+
+    return loadConfig(
+      {cwd: ctx.root, ...options},
+      mergeConfig(getDefaultConfig(ctx.root), overrideConfig),
     );
   }
 
+  // Load config file from disk, merging dynamic CLI config before project
+  // config using `defaultConfigOverrides` arg.
```

Comment:
Comment added to highlight this important `loadConfig` behaviour. Earlier versions of this PR made use of `mergeConfig`, which would be incorrect.